### PR TITLE
SALTO-956 - Move layout, custom object translation, record type filters to new format

### PR DESCRIPTION
**Pull request by netama**
Replace existing filters with reference rules + fix deploy translation by marking the ones that were missed as relative.
Also includes some small parts from the next steps of SALTO-942 - I ran into the missing relative translations while working on another extension, so I removed the completely-new functionality and will add it in another PR, but didn't want to revert the name and format changes that could apply here too.

A note about testing: I verified each of the items manually on a test account, and also converted the tests I could before removing them. You can look at the latest commit's changes to see the `record_type` example (some of the others required too many adjustments to the original tests due to missing types, so I only verified them manually). For the new format, since these are all "standard" cases in the new configuration, I think unittests are not the best way to test them individually. I'll add a note on the ticket (SALTO-942) with some suggested approaches.

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/field_references.ts [hunk 1]
```diff
@@ -28,25 +28,25 @@ import { parentApiName } from './utils'
...
-type ContextFunc = (instance: InstanceElement, ...strategyArgs: string[]) => string[]
 const ContextStrategyLookup: Record<
   ReferenceContextStrategyName, ContextFunc
 > = {
-  none: () => [],
+  none: () => undefined,
   instanceParent: instance => (isDefined(instance.annotations[INSTANCE_ANNOTATIONS.PARENT])
-    ? [parentApiName(instance)]
-    : []
+    ? translateSpecialCases(parentApiName(instance))
```
Comment (ori-moisis) [inline packages/salesforce-adapter/src/filters/field_references.ts:48]: why is this here?
seems like duplicate logic to finding the parent...
if there is already a `_parent` annotation, then this translation was already done in the layout filter, right?
we should be able to use `apiName(instance.annotations[INSTANCE_ANNOTATIONS.PARENT])` (maybe needs a `[0]` at the end, I don't remember whether _parent is a list or not) instead of re-calculating the parent name here

**Review by ori-moisis — APPROVED** [review]

File: packages/salesforce-adapter/src/filters/field_references.ts (hunk 1 above)
Comment (netama) [reply packages/salesforce-adapter/src/filters/field_references.ts:48]: It's more complex than that because we can't use `apiName` on the parent annotation (which is an `ElemID`), we have to first find the element by its elemid, and we didn't need that mapping until this point.
Please lmk if it's ok to merge this or if I should add a by-elemid lookup for this filter

Comment (netama) [reply packages/salesforce-adapter/src/filters/field_references.ts:48]: merging for now to address the original bug - will make the adjustments in a follow-up PR
